zonal_datacube/zonal_datacube.py

```diff
@@ -0,0 +1,262 @@
+
+            for _, zone in zones_per_tile.iterrows():
+                zone_geometry = wkt.loads(zone.zone_wkt)
+                geom_masked_datacube = ZonalDataCube._mask_datacube_by_geom(
+                    zone_geometry, masked_datacube, tile.bounds
+                )
+
+                zone_attributes = zone.drop("zone_wkt")
+                result = zone_attributes.copy()
+                for column, func_spec in analysis_funcs.items():
+                    func_result = func_spec.func(zone_attributes, geom_masked_datacube)
+
+                    if isinstance(func_result, pd.Series):
+                        result = pd.concat([result, func_result])
+                    else:
+                        result[column] = func_spec.func(
+                            zone_attributes, masked_datacube
+                        )
+
+                partition_results.append(result)
+
+        return pd.DataFrame(partition_results)
+
+    @staticmethod
+    def _get_dask_zones(zones, bounds, cell_size, npartitions):
+        # fishnet features to make them partition more efficiently in Dask
+        fishnetted_zones = fishnet(zones, *bounds, cell_size)
+
+        # Dask can't understand the geometry type,
+        # so just serialize to WKT and drop geometry
+        fishnetted_zones["zone_wkt"] = fishnetted_zones["geometry"].apply(
+            lambda x: x.wkt
+        )
+        del fishnetted_zones["geometry"]
+
+        dask_zones = dd.from_pandas(fishnetted_zones, npartitions=npartitions)
+
+        # Index based on the fishnet geometry so features
+        # that overlap the same fishnet grid cell
+        # are processed in the same partition.
+        # This ensures we only need to read the raster data
+        # only once per grid cell across the cluster
+        indexed_zones = dask_zones.set_index("fishnet_wkt", npartitions=npartitions)
+
+        return indexed_zones
+
+    @staticmethod
+    def _get_meta(zones, analysis_funcs):
+        meta = dd.utils.make_meta(zones)
+
+        # drop geom columns for meta
+        if "zone_wkt" in meta:
+            meta = meta.drop("zone_wkt", axis=1)
+        elif "fishnet_wkt" in meta:
+            meta = meta.drop("fishnet_wkt", axis=1)
+
+        for col, func in analysis_funcs.items():
+            if func.meta:
+                for col, type in func.meta.items():
+                    meta[col] = pd.Series(dtype=type)
+            else:
+                # TODO how do we determine dtype if they didn't specify meta?
+                meta[col] = pd.Series(dtype="float64")
+
+        return meta
+
+    @staticmethod
+    def _get_default_analysis_funcs(
+        analysis_funcs: List[str], stac_items: List[pystac.Item]
+    ) -> Dict:
+        return {
+            func: get_default_analysis_function(func, stac_items)
+            for func in analysis_funcs
+        }
+
+    def _get_stac_items(self, stac_items, spatial_only):
```

Comment:
I think this got buried within the other static methods. Maybe just pull it up closer to the other instance methods?